lsp-go.el

```diff
@@ -182,38 +154,29 @@ $GOPATH/pkg/mod along with the value of
         (mapcar (lambda (path) (concat (file-remote-p default-directory) path)) library-dirs)
       library-dirs)))
 
-(define-inline lsp-clients-go--bool-to-json (val)
-  (inline-quote (if ,val t :json-false)))
-
-(defun lsp-clients-go--make-init-options ()
-  "Init options for golang."
-  `(:funcSnippetEnabled ,(lsp-clients-go--bool-to-json lsp-clients-go-func-snippet-enabled)
-                        :disableFuncSnippet ,(lsp-clients-go--bool-to-json (not lsp-clients-go-func-snippet-enabled))
-                        :gocodeCompletionEnabled ,(lsp-clients-go--bool-to-json lsp-clients-go-gocode-completion-enabled)
-                        :formatTool ,lsp-clients-go-format-tool
-                        :goimportsLocalPrefix ,lsp-clients-go-imports-local-prefix
-                        :maxParallelism ,lsp-clients-go-max-parallelism
-                        :useBinaryPkgCache ,lsp-clients-go-use-binary-pkg-cache
-                        :diagnosticsEnabled ,lsp-clients-go-diagnostics-enabled))
+(defcustom lsp-go-link-target "pkg.go.dev"
+  "Which website to use for displaying Go documentation."
+  :type '(choice (const "godoc.org")
+		 (const "pkg.go.dev")
+		 (string :tag "A custom website"))
+  :group 'lsp-go
+  :package-version '(lsp-mode "6.3.2"))
 
+(lsp-register-custom-settings
+ '(("gopls.usePlaceholders" lsp-go-use-placeholders t)
+   ("gopls.hoverKind" lsp-go-hover-kind)
+   ("gopls.buildFlags" lsp-go-build-flags)
+   ("gopls.env" lsp-go-env)
+   ("gopls.linkTarget" lsp-go-link-target)))
 
 (lsp-register-client
  (make-lsp-client :new-connection (lsp-stdio-connection
-                                   (lambda () (cons lsp-clients-go-server
-                                                    lsp-clients-go-server-args)))
+                                   (lambda () (cons lsp-go-gopls-server-path lsp-go-gopls-server-args)))
                   :major-modes '(go-mode)
-                  :priority -1
-                  :initialization-options 'lsp-clients-go--make-init-options
-                  :server-id 'go-bingo
-                  :library-folders-fn 'lsp-clients-go--library-default-directories))
-
-(lsp-register-client
- (make-lsp-client :new-connection (lsp-stdio-connection "go-langserver")
-                  :major-modes '(go-mode)
-                  :priority -2
-                  :initialization-options 'lsp-clients-go--make-init-options
-                  :server-id 'go-ls
-                  :library-folders-fn 'lsp-clients-go--library-default-directories))
+		  :language-id "go"
```

Comment:
The indentation seems not correct here.

Reply:
Yes, something's off with my indentation config. Let me see.